### PR TITLE
refactor: Migrate to named exports

### LINKED ISSUE #5153: Migrate to named exports
### Background

The [JavaScript](https://google.github.io/styleguide/jsguide.html#es-module-exports) and [TypeScript](https://google.github.io/styleguide/tsguide.html#exports) style guides specifically prohibit default exports for ES/TS Modules:

> ```JS
> // Do not use default exports:
> export default class Foo { ... } // BAD!
> 
> // Use named exports:
> export class Foo { ... }
> 
> // Alternate style named exports:
> class Foo { ... }
> export {Foo};
> ```

For `goog.module` modules, the JavaScript style guide does not specifically prohibit default exports (`exports = MyClass;`), but it gives only examples of named exports.  (It does give examples of `goog.require`ing the default export, however.)

### Proposal

Migrate all modules which have a default export to use only named exports, so

```JS
const MyClass = function(...) {...};
exports = MyClass
```
becomes
```JS
const MyClass = function(...) {...};
exports.MyClass = MyClass;
```
and the corresponding import is (simultaneously) changed from
```JS
const MyClass = goog.require('MyClass');
```
to
```
const {MyClass} = goog.require('MyClass');
```

### Discussion

There's nothing wrong with default exports in the short term, but this is something we will want to fix eventually as we continue the migration from `goog.provide` -> `goog.module` -> ES modules -> TypeScript.  Issue #5073 records a number of cases where we will probably decide to fix this sooner rather than later.  This bug is to ensure that "later" arrives eventually.

### PR DESCRIPTION
**Pull request by gonfunko**
## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves
#5153

### Proposed Changes
Moves classes to named exports instead of default exports.

#### Behavior Before Change
Classes were exported as default exports.

#### Behavior After Change
Classes are exported as named exports.

### Reason for Changes
Styleguide conformance and work towards ES6/Typescript.

### Test Coverage
Verified all test suites are passing.

### CODE REVIEW

File: core/utils/svg.js [hunk 1]
```diff
@@ -181,4 +181,4 @@ Svg.TEXT = new Svg('text');
  */
 Svg.TSPAN = new Svg('tspan');
 
-exports = Svg;
+exports.Svg = Svg;
```
Comment (cpcallen) [inline core/utils/svg.js:184]: I'm a bit sceptical about this one: shouldn't all of the `Svg.*` constants be top-level named exports of this module?

Of course, it might be best to address that in a separate PR, so I've added this to the relevant list in #5073.

**Review by cpcallen — CHANGES_REQUESTED** [review]
Can you add entries to `scripts/migration/renamings.js` for these (in some automated way)?  Although they won't affect users who use the main `blockly` package, they will affect projects that import modules individually.

Create a new top-level entry for post-Q3-release changes:
```JS
  '7.20211209.0-beta.0': {
```
and then add entries of the following form for each module with a newly-named formerly-default export:
```
    'Blockly.whatever.ClassName': {
      export: 'ClassName',
      path: 'Blockly.whatever.ClassName',
    },
```
…except if the module is not reexported by `blockly.js` (even indirectly, e.g. via `Blockly.utils`), in which case the entry should omit the `path` property.

**Review by rachel-fenichel — APPROVED** [review]
Approved after internal discussion. Any additional fixes can be done in follow-ups, to avoid leaving this 235-file PR open and blocking other changes.

File: core/dropdowndiv.js [hunk 2]
```diff
@@ -759,6 +759,7 @@ DropDownDiv.repositionForWindowResize = function() {
     DropDownDiv.hide();
   }
 };
-exports = DropDownDiv;
 
-exports.TEST_ONLY = internal;
+DropDownDiv.TEST_ONLY = internal;
+
+exports.DropDownDiv = DropDownDiv;
```
[the remark this answers is not in the document]
Comment (cpcallen) [reply core/dropdowndiv.js:765]: Are you sure?  It appears that the corresponding change was made in `blockly.js`:

```JS
const {DropDownDiv} = goog.require('Blockly.DropDownDiv');
// ...
exports.DropDownDiv = DropDownDiv;
```
